Feat: add timeout

parthpnx opened the pull request:
## Overview
Adding a timeout to skprmail avoidingfor unnecessarily long wait times with no errors or closures
## Implementation
There was a context created and passed to both the ses and the default method. The ses method automatically handles the timeout via the context whereas we've implemented a goroutine to check for context timeout and close the application.
## How to Test
Setup a mail server using `smtp-go` (https://github.com/emersion/go-smtp/blob/master/cmd/smtp-debug-server/main.go) package and add sleep at `/server.go:211` This will ensure that the smtp server is available but will not respond to a connection dial leading our skprmail to be hung until the timeout is hit.
## What is the rollout plan?
Merge and rollout
## Does this change need a blog post?
Maybe a blog pose around contexts and timeouts
## Link to Tickets
* <not sure>

parthpnx committed:
feat: adds timeout to both sendmail methods

parthpnx committed:
chore: update timeout for testing

parthpnx committed:
fix: timeout goroutine to exit program

parthpnx committed:
fix: goroutine to manage context timeout

parthpnx committed:
fix: linting error

nickschuch commented on internal/provider/default/default.go:
```diff
@@ -16,13 +18,13 @@ const (
 	EnvFrom = "SKPRMAIL_FROM"
 	// FallbackAddr where mail will be forwarded to.
-	FallbackAddr = "mail:1025"
+	FallbackAddr = "localhost:1025"
```
This should still be `mail`

nickschuch commented on internal/provider/default/default.go:
```diff
@@ -16,13 +18,13 @@ const (
+	FallbackAddr = "localhost:1025"
 	// FallbackFrom address which will be applied to email.
-	FallbackFrom = "skprmail"
+	FallbackFrom = "skprmail@skpprmail.com"
```
skprmail@localhost ?

nickschuch commented on internal/provider/default/default.go:
```diff
@@ -38,16 +40,57 @@ func Send(to []string, msg *mail.Message) error {
+
+	go func() {
+		for {
+			time.Sleep(1 * time.Second)
```
This whole thing could just become a "sleep for 30 seconds, then exit"

parthpnx committed:
fix: refactor code & add timeoutflag for binary
- adds smtp debug server binary for testing timeout
- update documentation for testing

parthpnx committed:
fix: add the go-smtp package instead of binary to run comre customization to the test smtp server.
